pontoon/base/signals.py

```diff
@@ -8,13 +8,48 @@
 
 from pontoon.base import errors
 from pontoon.base.models import (
+    Entity,
     Locale,
     Project,
     ProjectLocale,
     TranslatedResource,
     UserProfile,
 )
 
+from pontoon.base.utils import (
+    get_singulars
+)
+
+@receiver(pre_save, sender=Entity)
+def entity_singular_forms(sender, **kwargs):
+    """
+    Generate entity singular forms in order to speed-up processing.
+    """
+    instance = kwargs.get('instance', None)
+
+    if instance.string:
+        instance.string_singulars = ' '.join(get_singulars(instance.string))
+    else:
+        instance.string_singulars = ''
+    if instance.string_plural:
+        instance.string_plural_singulars = ' '.join(get_singulars(instance.string_plural))
+    else:
+        instance.string_plural_singulars = ''
```

Comment:
I'll double check what was the reason, because you're probably right -> that's not needed anymore.